### ISSUE TITLE
InventoryManager.CheckExpirations catch

### ISSUE THREAD

**Issue by jorbor** [MadTigger/Valant#2]
Is there a way you could log errors and return error messages to the consumer without catching here?

**Comment by MadTigger** [comment]
Certainly, there are a number of patterns for managing exceptions. I only trap errors here since this is code written in isolation and I don't know what the larger story is for handling errors. In all likelihood, there is probably going to be system is already in place to manage exceptions, and there is no point in re-inventing the wheel. It is often good form to let exceptions bubble up quite a bit to capture the full call stack and context that the error occurred in.
